kornia/geometry/epipolar/projection.py

```diff
@@ -107,6 +107,44 @@ def projection_from_KRt(K: torch.Tensor, R: torch.Tensor, t: torch.Tensor) -> to
 
     return K @ Rt
 
+def KRt_from_projection(P: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+    r"""Get the intrinsics, rotation-matrix and the camera-center.
+
+    This function decomposes the Projection matrix into Camera-Matrix, Rotation Matrix and Translation vector.
+
+
+    Args:
+        P (torch.Tensor): the projection matrix with shape :math:`(B, 3, 4)`.
+
+    Returns:
+        torch.Tensor: The Camera matrix with shape :math:`(B, 3, 3)`.
+        torch.Tensor: The Rotation matrix with shape :math:`(B, 3, 3)`.
+        torch.Tensor: the Translation vector with shape :math:`(B, 3)`.
+
+    """
+
+    assert P.shape[-2:] == (3, 4), "P must be of shape [B, 3, 4]"
+
+    submat_3x3 = P[:, 0:3, 0:3]
+    last_column = P[:, 0:3, 3].unsqueeze(-1)
+    t = torch.matmul(-torch.inverse(submat_3x3), last_column).squeeze(-1)
+
+    # Trick to turn QR-decomposition into RQ-decomposition
+    reverse = torch.Tensor([[0, 0, 1], [0, 1, 0], [1, 0, 0]]).unsqueeze(0)
```

Comment:
you need to pass `device` and `dtype` otherwise some tests won't pass
`reverse = torch.tensor([[[0, 0, 1], [0, 1, 0], [1, 0, 0]]], device=P.device, dtype=P.dtype)`